clients/src/main/java/org/apache/kafka/common/security/authenticator/SaslServerAuthenticator.java

```diff
@@ -73,6 +73,7 @@
 
 public class SaslServerAuthenticator implements Authenticator {
 
+    private static final int MAX_RECEIVE_SIZE = 131072; // Chosen as twice the max receive size for GSSAPI
```

Comment:
ok, makes sense to make the changes only on server-side now. And 512K sounds reasonable to me.